modules/launch.nix

```diff
@@ -50,7 +50,18 @@ let
   launcher = pkgs.callPackage ../launcher {};
   dbusOutsidePath = concat (env "XDG_RUNTIME_DIR") (concat "/nixpak-bus-" instanceId);
   
+  pastaEnable = config.bubblewrap.network && config.pasta.enable;
+  originalBwrapExe = "${config.bubblewrap.package}/bin/bwrap";
+  bwrapExe = if pastaEnable then "${config.pasta.package}/bin/pasta" else originalBwrapExe;
```

Comment:
This feels pretty hacky. It also seems to break the xdg-dbus-proxy wrapper currently. We also need to make sure that wrapping bwrap doesn't break the sync fd shenanigans.

Reply:
I agree with you that it is not the best solution. However, it was the one that I could get quickly working.

It might be possible to avoid having pasta as additional wrapper. However, this would require some major changes in the nixpak launcher.
Currently, it just sets up everything and hands over to bwrap by exec'ing it.
If we want to go down the other route, we would need bwrap to first setup the sandbox, then have the launcher adjust the network namespace with pasta before then bwrap actually executes the application within the sandbox.
This might be possible by using bwrap's `--block-fd` and pasta's *PID* option. However, we might have to somehow magically find the PID of the subprocess that bwrap forks (not sure if we can get that directly from bwrap or if we have to rely on the proc filesystem).

I think we might need to change the nixpak launcher anyway to something supervising the sandbox and, therefore, running while the whole sandbox lifetime.
I am working on a few more sandboxing features, namely for Wayland using `wayland-proxy-virtwl` and for PipeWire using `pw-container`.
Both tools have to run for the whole sandbox lifetime and have no mechanism like `xdg-dbus-proxy` to exit when an FD is closed.
I have basically both of these sandboxing features already working, but they currently leave dangling processes when the application exists.

What do you think about changing nixpak's launcher into a sandbox supervisor?
Would you prefer running pasta separately instead of as wrapper?

Reply:
If you want to take a sneak peak at my current state of Wayland and PipeWire sandboxing features: https://github.com/lxp/nixpak/pull/1
On my system, I am also running a re-implemented version of Snap's PipeWire access control scripts. However, I am not sure, if these changes can be integrated into nixpak, since they require changes to existing system services on the host system.

Reply:
I moved the pasta command construction into the Go launcher to fix the most obvious bug that the `--sync-fd` parameter is passed to pasta instead of bwrap.
However, it seems that pasta closes the sync fd used for xdg-dbus-proxy :(
I think the only way to proceed is to make nixpak's launcher into a sandbox supervisor and implement pasta support without wrapping the bwrap command like I described in my previous comment.
If you have different ideas, please let me know.